toolkit/smart-contracts/offchain/src/governance.rs

```diff
@@ -252,6 +252,41 @@ pub async fn get_governance_policy_summary<T: QueryLedgerState + QueryNetwork>(
 	}
 }
 
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct MultiSigParameters {
+	governance_authorties: Vec<MainchainKeyHash>,
+	threshold: u8,
+}
+
+impl MultiSigParameters {
+	pub fn new(governance_authorties: &[MainchainKeyHash], threshold: u8) -> Result<Self, &str> {
+		if governance_authorties.is_empty() {
+			return Err("governance authorities cannot be be empty");
```

Comment:
we have few errors, what about some thiserror?

Reply:
To be honest, I wish this crate was implemented with `thiserror`, but we went with `anyhow`. Having said that, should I make it `anyhow`? I think introducing `thiserror` should be a standalone task for this crate. It would also be a breaking change, so I'm a bit reluctant now. Maybe for "2.0".